src/evaluation/Run.jl

```diff
@@ -4,6 +4,54 @@ import .ExpressionExplorer: FunctionNameSignaturePair, is_joined_funcname, Using
 
 Base.push!(x::Set{Cell}) = x
 
+"""
+Recursively checks which downstream dependencies need to be deactivated
+"""
+function get_deactivated_cell_uuids!(deactivated_uuids:: Set{UUID}, cell:: Cell)
+	cell.cell_id ∈ deactivated_uuids && return # if a cell is already deactived, all its downstream dependencies are also already processed
+	push!(deactivated_uuids, cell.cell_id)
+	references = cell.cell_dependencies.downstream_cells_map
+    isempty(references) && return
+
+    referenced_cells = vcat(values(references)...) |> unique
+    for c ∈ referenced_cells
+        get_deactivated_cell_uuids!(deactivated_uuids, c)
+    end
+	nothing
+end
+
+"""
+Get UUIDs for all deactivated cells (directly or indirectly)
+"""
+function get_deactivated_cell_uuids(cells:: Vector{Cell}):: Set{UUID}
+	deactivated_uuids = Set{UUID}()
+	for cell in cells
+		if cell.has_execution_barrier
+			get_deactivated_cell_uuids!(deactivated_uuids, cell)
+		end
+	end
+	return deactivated_uuids
+end
+
+"""
+Deactivation of cells for execution barriers.
+"""
+function cell_deactivation!(cells_in:: Vector{Cell}, deactivated_uuids:: Set{UUID})
+    # activate all cells before checking which cells are affected by execution barrier
+    for cell in cells_in
+        cell.is_deactivated = false
+    end
+    # identify cells affected by active execution barrier and its references
+    for cell in cells_in
+		if cell.cell_id ∈ deactivated_uuids
+			cell.is_deactivated = true
+			cell.queued = false
```

Comment:
why `queued=false`?

Reply:
My original idea is that deactived cells could not be queued for execution, therefore it is safer to set it explicitly to false. But not sure if this is really needed.